standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/tools/schematool/TestSchemaToolForMetastore.java

```diff
@@ -467,6 +471,18 @@ public void testMetastoreDbPropertiesAfterUpgrade() throws HiveMetaException, IO
     validateMetastoreDbPropertiesTable();
   }
 
+  @Test
+  public void testValidateHiveShortVersion() {
+    String hiveShortVersion = MetastoreVersionInfo.getShortVersion();
+    String hiveVersion = MetastoreVersionInfo.getVersion();
+    Assert.assertEquals(hiveVersion, StringUtils.join(hiveShortVersion, "-SNAPSHOT"));
+    Assert.assertTrue(hiveVersion.startsWith(hiveShortVersion));
+    String fileName = Stream.of("src/test/resources/sql/postgres/upgrade-3.1.3000-to-" , hiveShortVersion ,".postgres.sql").
+            collect(Collectors.joining());
+    File file = new File(fileName);
+    Assert.assertTrue(file.exists());
```

Comment:
I think that this test may be a bit redundant since if hiveShortVersion is different, the metastore upgrade will fail when the relevant tests run.

I like the idea of having unit tests checking that upgrade and install scripts are present for the current version but if we go into this direction we should be better doing that for all supported databases and in a more appropriate place (not inside this class).

Reply:
This assertion tried to avoid PostgresTPCDS rule failure because of wrong file lookup of upgrade scripts. But agree to the point that we need to have it for all DBs with install and upgrade scripts where  we need to add all install upgrade scripts under sql/db/.. . Probably seprate jira ticket?

Please suggest if I can add these assertions for all dbs in new class something like ->TestDbInstallBase . 

PS- there is rule for mariadb but no test for it . Even if adding test needs further investigation as it fails with mariaDb 10.2 image.

Reply:
If there are no tests that are able to guarantee that the scripts exist and are working as expected then we should check how to improve code coverage. At the moment, I am not sure we are missing something. If we do then yes log a JIRA ticket and add your suggestions.

The MariaDB failure is logged under https://issues.apache.org/jira/browse/HIVE-27749. Feel free to pick this up and fix it if you feel like it.